spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageProducingHandler.java

```diff
@@ -134,22 +140,26 @@ private void updateNotPropagatedHeaders(String[] headers, boolean merge) {
 			this.notPropagatedHeaders.addAll(Arrays.asList(headers));
 		}
 		this.selectiveHeaderPropagation = this.notPropagatedHeaders.size() > 0;
+
+		this.shouldCopyRequestHeaders = !this.notPropagatedHeaders.contains("*");
```

Comment:
That's not. Because `notPropagatedHeaders` is a `Set` not simple `String`. So, we don't copy headers at all if one of the `notPropagatedHeaders` elements is exactly asterisk, e.g. `foo, b*, *, b*z`.
So, all other patterns will be ignored just because we have one of them as `*`. It has a highest priority.

Reply:
Of course, sry!